registry/server/templates/routes/getTemplates.ts

```diff
@@ -1,15 +1,64 @@
-import { Request, Response } from 'express';
+import { Response } from 'express';
 
 import db from '../../db';
 import Template from '../interfaces';
 import { Tables } from '../../db/structure';
 import { appendDigest } from '../../util/hmac';
 import { EntityTypes } from '../../versioning/interfaces';
+import Joi from 'joi';
+import { filtersMiddleware, RequestWithFilters } from '../../middleware/filters';
+import { ok } from 'assert';
+
+interface Filters {
+    domainId?: number | 'null';
+    id?: string[] | string;
+    name?: string[] | string;
+}
+
+const filtersSchema = Joi.object<Filters>({
+    domainId: Joi.alternatives(Joi.number(), Joi.string().valid('null')).optional(),
+    id: Joi.alternatives(Joi.string(), Joi.array().items(Joi.string())).optional(),
+    name: Joi.alternatives(Joi.string(), Joi.array().items(Joi.string())).optional(),
+});
+
+const getTemplates = async (req: RequestWithFilters<Filters>, res: Response): Promise<void> => {
+    const query = db
+        .selectVersionedRowsFrom<Template>(Tables.Templates, 'name', EntityTypes.templates, [`${Tables.Templates}.*`])
+        .orderBy('name', 'asc');
+
+    if (req.filters?.id || req.filters?.name) {
+        const name = req.filters?.name ?? req.filters?.id;
+        ok(name);
+        const normalizedNames = Array.isArray(name) ? name : [name];
+        query.whereIn('name', normalizedNames);
+    }
+
+    if (req.filters?.domainId) {
```

Comment:
matching for template500 seems to happen multiple times, utilize some constant for that?

Reply:
It's a field name, just like "name" or "domainId". If we put it into constant, we will need to do the same for the rest of the fields, just like we do for tables.